src/request-pipeline/index.js

```diff
@@ -136,9 +136,17 @@ function isDestResBodyMalformed (ctx) {
 
 // API
 export function run (req, res, serverInfo, openSessions) {
-    var ctx = new RequestPipelineContext(req, res, serverInfo);
-
-    if (ctx.dispatch(openSessions)) {
+    var ctx         = new RequestPipelineContext(req, res, serverInfo);
+    var isDispatch  = ctx.dispatch(openSessions);
+    var servicePage = ctx.getServicePage();
+
+    if (isDispatch && servicePage) {
+        res.setHeader('content-type', 'text/html');
+        res.statusCode = 200;
+        res.write('<html><body><h1>' + servicePage.name + '</h1></body></html>');
+        res.end();
```

Comment:
Wrong. We should check parsed url for `about:blank' or 'about:error'. Set content to empty string and directly jump to the step `5` of pipeline.